syntax = "proto3";

// an iKAT run has run_name and run_type fields plus
// a list of Turns
message iKATRun {
    string run_name = 1;
    string run_type = 2;
    bool eval_response = 4;
    repeated Turn turns = 3;
}

// each Turn has a string ID and one or more Responses
message Turn {
    string turn_id = 1;
    repeated Response responses = 2;
}

// each Response has an integer rank, the text response
// to give to the user, a list of passage provenances,
// and a list of PTKB provenances
message Response {
    string text = 1;
    int32 rank = 2;
    repeated PassageProvenance passage_provenance = 3;
    // unlike 2023, this field is now just a list of 
    // PTKB entry IDs
    repeated int32 ptkb_provenance = 4;
}

// PassageProvenances contain passage ID, 
// passage text, passage score
message PassageProvenance {
    string id = 1;
    string text = 2;
    float score = 3;
    bool used = 4;
}
